backends/dpdk/dpdkHelpers.cpp

```diff
@@ -134,27 +135,64 @@ bool ConvertStatementToDpdk::preorder(const IR::AssignmentStatement *a) {
+                    auto consOrgBitwidth = src2Op->to<IR::Expression>()->type->width_bits();
+                    auto bitWidth = consOrgBitwidth < 32 ? 32 : 64;
+                    if (src2Op->is<IR::Constant>()) {
+                        auto consValue = src2Op->to<IR::Constant>()->value;
+                        auto val = consValue << (bitWidth - consOrgBitwidth);
+                        consValue = val >> (bitWidth - consOrgBitwidth);
+                        src2Op = new IR::Constant(IR::Type_Bits::get(bitWidth), consValue);
+                    } else {
+                        auto val = bitWidth - consOrgBitwidth;
+                        add_instr(new IR::DpdkShlStatement(src2Op, src2Op, new IR::Constant(val)));
+                        add_instr(new IR::DpdkShrStatement(src2Op, src2Op, new IR::Constant(val)));
+                    }
+                }
+            }
+        }
+
         if (right->is<IR::Add>()) {
-            i = new IR::DpdkAddStatement(left, src1Op, src2Op);
+            add_instr(new IR::DpdkAddStatement(left, src1Op, src2Op));
         } else if (right->is<IR::Sub>()) {
-            i = new IR::DpdkSubStatement(left, src1Op, src2Op);
+            add_instr(new IR::DpdkSubStatement(left, src1Op, src2Op));
         } else if (right->is<IR::Shl>()) {
-            i = new IR::DpdkShlStatement(left, src1Op, src2Op);
+            add_instr(new IR::DpdkShlStatement(left, src1Op, src2Op));
         } else if (right->is<IR::Shr>()) {
-            i = new IR::DpdkShrStatement(left, src1Op, src2Op);
+            add_instr(new IR::DpdkShrStatement(left, src1Op, src2Op));
         } else if (right->is<IR::Equ>()) {
-            i = new IR::DpdkEquStatement(left, src1Op, src2Op);
+            add_instr(new IR::DpdkEquStatement(left, src1Op, src2Op));
         } else if (right->is<IR::LOr>() || right->is<IR::LAnd>()) {
             process_logical_operation(left, r);
         } else if (right->is<IR::BOr>()) {
-            i = new IR::DpdkOrStatement(left, src1Op, src2Op);
+            add_instr(new IR::DpdkOrStatement(left, src1Op, src2Op));
         } else if (right->is<IR::BAnd>()) {
-            i = new IR::DpdkAndStatement(left, src1Op, src2Op);
+            add_instr(new IR::DpdkAndStatement(left, src1Op, src2Op));
         } else if (right->is<IR::BXor>()) {
-            i = new IR::DpdkXorStatement(left, src1Op, src2Op);
+            add_instr(new IR::DpdkXorStatement(left, src1Op, src2Op));
         } else {
             BUG("%1% not implemented.", right);
         }
+        /* If there is a binary operation present which involves non 8-bit aligned
+           fields , add BAnd operation with mask value of original bit-width to
+           avoid any result overflow */
+        if (!isEightBitAligned(left) && !isSignExtended) {
+            if (right->is<IR::Add>() || right->is<IR::Sub>() ||
+                right->is<IR::Shl>() || right->is<IR::Shr>()) {
+                auto width = left->type->width_bits();
```

Comment:
Maybe you want to use the Constant class, this may overflow in C++.

Reply:
Sorry. I did not get the comment. Could you please add more to understand?